gtfu/src/main/java/gtfu/GraphicReport.java

```diff
@@ -171,21 +176,26 @@ public GraphicReport(String cacheDir, String selectedDate, String savePath, bool
 
             // Only create report for agencies with trip report data
             if (tdList.size() > 0) {
-                blobs.add(imageToBlob(img));
+                // converts <agency-id>-yyyy-mm-dd.txt to <agency-id>-yyyy-mm-dd
+                blobMap.put(key.substring(0, key.length() - 4), imageToBlob(img));
             }
         }
-        if (downloadReport) {
```

Comment:
It doesn't seem like `downloadReport` really has a purpose anymore. Either `savePath` is `null`, which means don't download, or it has a value, which means download.